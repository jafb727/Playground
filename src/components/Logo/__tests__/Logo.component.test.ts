/*
 * @author: Jose A Felix
 * @editor: Jose A Felix
 * @name: Logo.component.test
 * @description: Logo component test cases
 */

/* --------------------------------------------- */

/** @import Libraries */
import { act, screen } from "@testing-library/react";

/** @import Components */
import Component from "../Logo.component";

/** @import Utilities */
import { basicSetup } from "../../../test/setupTests";

/* --------------------------------------------- */

describe("Logo.component", () => {
   /** @constant Properties */
   const props = {
      options: {
         altText: "Login form logo",
         alignment: "center",
         size: "small",
         color: "primary",
      },
   };

   test("Renders component ok", async () => {
      basicSetup(Component, props);
      const component = screen.getByTestId("logo");

      await act(() => {
         expect(component).toBeInTheDocument();
      });
   });

   /* ----------------------- */

   test("Passes shallow snapshot test", async () => {
      const component = basicSetup(Component, props);

      await act(() => {
         expect(component).toMatchSnapshot();
      });
   });
});
