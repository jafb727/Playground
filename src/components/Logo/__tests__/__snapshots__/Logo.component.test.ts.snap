// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo.component Passes shallow snapshot test 1`] = `
<div
  class="logoContainer  MuiBox-root css-nntn6c"
  data-testid="logo"
>
  <div
    class="imageLogoContainer MuiBox-root css-0"
  >
    <img
      alt="Login form logo"
    />
  </div>
</div>
`;
